src/main/java/mezz/jei/config/ConfigValues.java

```diff
@@ -25,4 +25,7 @@
 	public boolean cheatItemsEnabled = false;
 	public boolean hideModeEnabled = false;
 	public String filterText = "";
+
+	// Sort
+	public String itemSortlist = ""; // Gets populated dynamically.
```

Comment:
I've considered doing that.  Mostly, I haven't done it because editing the config with that style with the default GUI does not allow simply moving entries up and down, so you are forced to add blank lines, then copy/cut one option and paste it into the new spot.  I can try it if you really would like it, play around with the sample ones in forge's config area.